trieste/acquisition/function/function.py

```diff
@@ -104,14 +105,14 @@ class ExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
     the maximum of the posterior mean at all observed points.
     """
 
-    def __init__(self, search_space: Optional[SearchSpace] = None, log_transform: bool = False):
+    def __init__(self, search_space: Optional[SearchSpace] = None):
         """
         :param search_space: The global search space over which the optimisation is defined. This is
             only used to determine explicit constraints.
         :param log_transform: If `True`,  use logEI proposed by :cite:`ament2023unexpected`.
         """
         self._search_space = search_space
-        self._log_transform = log_transform
+        self._acq_function_cls = expected_improvement
```

Comment:
I think a typical way to do this would be to create a private method, smth like this:
```
def _prepare_acq_function(model, eta):
    return expected_improvement(model, eta)
```

which you would then overwrite, overwriting init is a bit unusual way to do this  I think...